### ISSUE TITLE
Improve debugging bzlmod missing modules

### ISSUE THREAD

**Issue by keith** [bazelbuild/bazel#17233]
### Description of the feature request:

When you get this bzlmod error:

```
ERROR: Error computing the main repository mapping: module not found in registries: yams@5.0.2
```

It can be unclear where the issue is coming from. In the simplest case it's coming from your own MODULE.bazel, but it's also possible that it comes from nested modules, especially if you're using `--override_module` to debug something.

I think in this case it would be nice to have some way to trace back where this is coming from, potentially with a stack trace.

Another interesting thing in the specific case I hit was my main MODULE.bazel defined a real version 5.0.3, so no matter what this version would have been ignored. It's probably too dangerous to ignore invalid versions like this but in my case ignoring it would have been fine.

### Which operating system are you running Bazel on?

macOS

### What is the output of `bazel info release`?

6.0.0

**Comment by brentleyjones** [comment]
Not stale

**Comment by chrislovecnm** [comment]
While creating bzlmod extensions, I have found very few errors when an extension is not running.  The typical error that I receive is this:

```
@[unknown repo 'whl_mods_hub' requested from @rules_python~override]
```

This error states that a repository is not created or loaded correctly, but unfortunately, you do not have any information on debugging this error.  Moreover, the extension that I am working on is not even running.

I have the correct `use_repo` statement in the MODULE.bazel:

```
use_repo(whl_mods, "whl_mods_hub")
```

Then later, I'm using the `whl_mods_hub` repository in the call to another extension:

```
    "requests" : "@whl_mods_hub//:requests.json",
```

I have used the same pattern in other extensions.

I have also made other mistakes in the MODULE.bazel files and have not gotten any errors.  For instance, I misnamed
the module extension name, and did not get any errors.

```
whl_mods = use_extension("@rules_python//python/extensions:whl_mods.bzl", "whl_mods")
```

Should have been:

```
whl_mods = use_extension("@rules_python//python/extensions:whl_mods.bzl",  "whl_modifications")
```

I'm deducing that the extension is not running because I have a print statement at the top of the implementation of the module extension, and it is not printing anything.

Here is a branch for my work: https://github.com/chrislovecnm/rules_python/tree/wheel-mod-extension-not-working
